### PR TITLE
[SQL] Add toString to DataFrame/Column

### PR DESCRIPTION
**Pull request by marmbrus**

### CODE REVIEW

**Comment by marmbrus** [comment]
test this please

**Comment by marmbrus** [comment]
test this please

**Comment by marmbrus** [comment]
test this please

**Comment by marmbrus** [comment]
sigh... test this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala [hunk 1]
```diff
@@ -67,6 +68,17 @@ abstract class Expression extends TreeNode[Expression] {
   def childrenResolved = !children.exists(!_.resolved)
 
   /**
+   * Returns a string representation of this expression that does not have developer centric
+   * debugging information like the expression id.
+   */
+  def prettyString: String = {
+    transform {
```
Comment (scwf) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala]: why use transform here? this will change the expression

Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala]: We only replace some leaf nodes in the expression tree, does it cause problems?

Comment (scwf) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala]: no, i am going through the code, and i think we'd better not use transform here. in which case we need change the expression?

**Pull request closed without merging**